### ISSUE TITLE
ProtonVPN messes up with the DNS on my machine

### ISSUE THREAD

**Issue by ch33kaboo** [ProtonVPN/linux-app#117]
Context: I am using Manjaro Linux and downloaded the app from AUR (I used the metapackage and it installed everything for me).

Issue: When I open the app and try to connect to some servers (for example the Japan server shown in the picture), a window pops up and asks me for a password. I don’t even know what password it is asking me for, also the password field comes prefilled as shown in the image. When I forcefully kill that window (xkill for example), the DNS of my machine gets messed up, and I cannot connect to any website. The content of the file `/etc/resolve.conf` becomes `nameserver ::1` , only this line. 

eveytime this happens I get the dns working back again by changing the content of that file and restarting the NetworkManager service `sudo systemctl restart NetworkManager` and trying some other stuff I found on the internet I get it working again.

why am I getting this error and did other people encounter the same thing?

![protonVPN error](https://github.com/ProtonVPN/linux-app/assets/100171494/4f2b5e61-dc70-4181-be0b-5e70de56b18c)

**Comment by hekatonkheirex** [comment]
Yeah, is happening to me too. Already paid a month subscription, but the problem still there.
Also, I'm using plain Arch with the AUR package.

**Comment by bobsmitherton** [comment]
I, too, am experiencing something similar with my Proton VPN Linux GUI app. I have a paid subscription so I can pick and choose any server I please, but I have been getting the dreaded Authentication Required popup prompt a lot recently which won't allow me to connect. Sometimes trying other countries works and sometimes it doesn't. This most recent time it took 10+ minutes and a dozen tries before a successful connection. I even get the same issue on the Command Line Interface. Knock on wood, but I don't have the issue with my iPhone 13 Pro Max running iOS 16.6 and updated to the latest app version 4.2.3. Finally, I do not have my DNS settings altered as far as I can tell.

I am running on:

Kernel:                         Linux 6.2.0-26-generic (x86_64)
C Library:                     GNU C Library / (Ubuntu GLIBC 2.37-0ubuntu2) 2.37
Distribution:                 Ubuntu 23.04 
GNOME:                      44.3
Windowing System:     X11
Proton VPN GUI: 	     v1.12.0 (library: v3.14.0 / api-client: v0.7.1)
Proton VPN CLI:          v3.13.0

Both the GUI program and CLI were installed as per Proton's exact instructions located on their website.

Here is my screenshot:

![Screenshot from 2023-07-29 13-08-27](https://github.com/ProtonVPN/linux-app/assets/92812058/d1bf3eed-bf14-416b-9c60-b6703f2c4215)

Clicking "cancel" cancels and shows a message that reads: Unknown reason occured. and "connect" does nothing but keep popping up the same prompt.

**Comment by Barltarman** [comment]
Similar issue on Ubuntu 22.04.02 with the pre-release app, don't seem to have the same issue with DNS but constantly requiring credentials.

**Comment by flufficat** [comment]
I have also had the same issue on KDE Plasma version 5.27.6. This seems to be an issue with the ProtonVPN linux app itself because I have successfully used Mullvad VPN on the same setup. 

From my experimenting this is an issue with only certain vpn servers. I have made a list of servers that work for me and have resorted to using a script to select a random server from that list.

It appears to be a bug because it isks for the OpenVPN credentials which it already has access to. To reproduce this connect to a server that then asks for the credentials. Go to the network manager and look for the username and password under the OpenVPN connection for the server you are connected to. It already has filled in, the username and password that match those under the account settings in the protonvpn account page.

see also #13 this appears to be the same issue

**Comment by flufficat** [comment]
Update since I did a bit of digging around:
This has been fixed at least on my machine in the [protonvpn-nm-lib](https://github.com/ProtonVPN/protonvpn-nm-lib) [3.16.0](https://github.com/ProtonVPN/protonvpn-nm-lib/releases/tag/3.16.0) release. Despite mentioning this issue they have solved in the release notes, proton haven't commented on this issue so I thought I would.

I had to rebuild protonvpn once I had checked that version 3.16.0 was installed. I simply uninstalled it and then reinstalled it:
Uninstalling `protonvpn*` and installing `protonvpn`

Then it will be using the new version of the library. I would like to know if this has also fixed this issue on your machine as it seems to have completely solved it on mine. Best wishes
